src/lm_buddy/jobs/evaluation/hf_evaluate.py

```diff
@@ -89,8 +90,12 @@ def run_eval(config: HuggingFaceEvalJobConfig) -> Path:
     dataset = hf_dataset_loader.load_dataset(config.dataset)
 
     # Limit dataset length if max_samples is specified
-    if config.evaluation.max_samples is not None:
-        dataset = dataset.select(range(config.evaluation.max_samples))
+    max_samples = config.evaluation.max_samples
+    if max_samples is not None and max_samples > 0:
```

Comment:
```suggestion
    if max_samples and max_samples > 0:
```
we can simplify to check for truthiness here, i.e. checking for both not None and not 0